Albums/Albums/AlbumDetailTableViewController.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  AlbumDetailTableViewController.swift
+//  Albums
+//
+//  Created by Enrique Gongora on 3/9/20.
+//  Copyright © 2020 Enrique Gongora. All rights reserved.
+//
+
+import UIKit
+
+class AlbumDetailTableViewController: UITableViewController, SongTableViewCellDelegate {
+    func addSong(with title: String, duration: String) {
+        if let song = albumController?.createSong(duration: duration, identifier: UUID().uuidString, name: title) {
+            tempSongs.append(song)
+        }
+        tableView.reloadData()
+        let indexPath = IndexPath(row: tempSongs.count, section: 0)
+        tableView.scrollToRow(at: indexPath, at: .middle, animated: true)
+    }
+    
+
+    // MARK: - IBOutlets
+    @IBOutlet weak var albumTextField: UITextField!
+    @IBOutlet weak var artistTextField: UITextField!
+    @IBOutlet weak var genreTextField: UITextField!
+    @IBOutlet weak var coverURLTextField: UITextField!
+    
+    // MARK: - IBActions
+    @IBAction func saveTapped(_ sender: UIBarButtonItem) {
+        guard let name = albumTextField.text, let artistName = artistTextField.text, let genres = genreTextField.text, let url = coverURLTextField.text else { return }
+        if let album = album {
+            albumController?.update(album: album, artist: artistName, coverArt: [], genres: [], identifier: UUID().uuidString, name: name, songs: tempSongs)
+        } else {
+            albumController?.createAlbum(artist: artistName, coverArt: [], genres: [], identifier: UUID().uuidString, name: name, songs: tempSongs)
+        }
```

Comment:
I see you unwrapped genres and coverArt, but didn't use them in the controller functions.
You could do something like this:
```
let genre = genres.components(separatedBy: ", ")
let coverURLs = url.components(separatedBy: ", ").compactMap({ URL(string: $0) })
```
And then you can use these values inside of the coverArt and genres var in the controller functions.